components/ILIAS/Administration/classes/class.ilAdministrationGUI.php

```diff
@@ -61,47 +62,47 @@
 */
 class ilAdministrationGUI implements ilCtrlBaseClassInterface
 {
-    protected ilObjectDefinition $objDefinition;
-    protected ilHelpGUI $help;
-    protected ilDBInterface $db;
-    public ilLanguage $lng;
-    public ilGlobalTemplateInterface $tpl;
-    public ilTree $tree;
-    public ilRbacSystem $rbacsystem;
-    public int $cur_ref_id;
-    public string $cmd;
-    public ilCtrl $ctrl;
-    protected string $admin_mode = "";
-    protected bool $creation_mode = false;
-    protected int $requested_obj_id = 0;
-    protected AdminGUIRequest $request;
-    protected ilObjectGUI $gui_obj;
+    private const array COMMANDS = ['forward', 'jump', 'jumpToPluginSlot', 'showTree'];
+
+    private readonly ilObjectDefinition $obj_definition;
+    private readonly ilHelpGUI $help;
+    private readonly ilLanguage $lng;
+    private readonly ilGlobalTemplateInterface $tpl;
+    private readonly ilTree $tree;
+    private readonly ilAccessHandler $access;
+    private ilRbacReview $rbac_review;
+    private ilObjUser $user;
+    private readonly ilCtrl $ctrl;
+    private readonly AdminGUIRequest $request;
+    private readonly GlobalScreen $global_screen;
     private readonly ilLogger $logger;
 
+    private int $cur_ref_id;
+    private string $cmd;
+    private string $admin_mode = "";
+    private bool $creation_mode = false;
+    private int $requested_obj_id = 0;
```

Comment:
AFAIK `$cmd` is unused and `$creation_mode` is only used in `executeCommand` and can be made a local variable instead.

The three below can be made readonly as they are only set in `__construct`.

```suggestion
    private readonly int $cur_ref_id;
    private readonly string $admin_mode;
    private int $requested_obj_id;
```